### ISSUE TITLE
Lua Script locked up computer

### ISSUE THREAD

**Issue by torridgristle** [SourMesen/Mesen#483]
I'm sure this wouldn't happen with a proper script since I don't know what I'm doing yet, but it seems like it shouldn't happen at all.
```
function update()
    local camX = emu.readWord(30, 0)
    emu.drawString(0,0,camX,000000,ffffff)
end

while(true) do
    update()
end
```

Clicking the run script button didn't give any errors, but it did lock up my computer and the HDD activity light stayed on until I shut it all down a few minutes later.

**Comment by SourMesen** [comment]
Lua scripts in Mesen are meant to work via a series of callback functions that allow you to interact with the emulation core.

e.g, this should do what you were trying to do :
```
function displayInfo()
    local camX = emu.readWord(30, 0)
    emu.drawString(0,0,camX,000000,ffffff)
end

emu.addEventCallback(displayInfo, emu.eventType.endFrame);
```

Your code will loop infinitely, calling update() on each loop, without ever letting the emulation continue.  Each time a draw function is called, a new draw command object is created, so in this particular case, you will end up using large amounts of memory, which will probably causing Windows to run out of RAM and use the swap file (which is probably the cause of the HDD activity)

There's a number of built-in scripts you can check with the "Built-in scripts" button in the toolbar at the top, if you want some examples.

**Comment by torridgristle** [comment]
Now that you mention it, the built-in scripts never loaded whenever I tried to open them, not sure why. I thought maybe I had to download them but that just now hit as me odd.

Also the color for the string had to use FFFFFF converted to dec as 16777215 in order to get white, otherwise it's black with FFFFFF, and that just seems silly.

It does work a lot better now though.

```
function displayInfo()
    local camX = emu.readWord(0x30, 0)
    emu.drawString(0,0,camX,000000,16777215)
end

emu.addEventCallback(displayInfo, emu.eventType.endFrame);
```

**Comment by torridgristle** [comment]
Now that you mention it, the built-in scripts never loaded whenever I tried to open them, not sure why. I thought maybe I had to download them but that just now hit as me odd.

The script works a lot better now, thanks.

```
function displayInfo()
    local camX = emu.readWord(0x30, 0)
    emu.drawString(0,0,camX,0x000000,0xFFFFFF)
end

emu.addEventCallback(displayInfo, emu.eventType.endFrame);
```

**Comment by SourMesen** [comment]
Ah yea, I just remembered that the appveyor builds have an issue with built-in scripts (they don't load up properly, like you described).  I'll take a look.

**Comment by SourMesen** [comment]
The latest appveyor build should load the built-in scripts properly now.

**Comment by SourMesen** [comment]
I added a configurable timeout (defaults to 1 second, which should be more than enough) for scripts (80c80181ba59e96d328bbd5bb2878209d01dfcd6) - if any callback (or the initial setup) of the script takes more than the specified amount of time, execution stops.

This prevents infinite loops, etc.  I also added a limit to the number of active draw commands (500k) to prevent a bad script from using gigantic amounts of memory by mistake.
